Tools/CISupport/ews-app/ews/views/results.py

```diff
@@ -82,26 +103,24 @@ def build_event(self, data):
         return HttpResponse('Saved data for change: {}.\n'.format(change_id))
 
     def step_event(self, data):
-        if not self.is_valid_result(data):
-            _log.warn('Incomplete step event data')
-            return HttpResponse('Incomplete data.')
         _log.info('Step event received, step-id: {}'.format(data['step_id']))
 
         Step.save_step(hostname=data['hostname'], step_id=data['step_id'], build_id=data['build_id'], result=data['result'],
                    state_string=data['state_string'], started_at=data['started_at'], complete_at=data['complete_at'])
         return HttpResponse('Saved data for step: {}.\n'.format(data['step_id']))
 
     def is_valid_result(self, data):
-        if data['type'] not in [u'ews-build', u'ews-step']:
-            _log.error('Invalid data type: {}'.format(data['type']))
+        if data['type'] not in ['ews-build', 'ews-step', 'ci-build']:
+            _log.error(f'Invalid data type: {data["type"]}')
             return False
 
-        required_keys = {u'ews-build': ['hostname', 'change_id', 'build_id', 'builder_id', 'builder_name', 'builder_display_name',
+        required_keys = {'ews-build': ['hostname', 'change_id', 'build_id', 'builder_id', 'builder_name', 'builder_display_name',
                                            'number', 'result', 'state_string', 'started_at', 'complete_at'],
-                         u'ews-step': ['hostname', 'step_id', 'build_id', 'result', 'state_string', 'started_at', 'complete_at']}
+                         'ews-step': ['hostname', 'step_id', 'build_id', 'result', 'state_string', 'started_at', 'complete_at'],
+                         'ci-build': ['build_id', 'builder_display_name', 'commit_hash', 'pr_number', 'result', 'url']}
```

Comment:
Oh, I see this is what ensures we have a pr number in https://github.com/WebKit/WebKit/pull/48253/files#r2216751022, right?